packages/core/core/src/loadParcelConfig.js

```diff
@@ -130,7 +131,18 @@ export function validateConfigFile(
   relativePath: FilePath
 ) {
   validateNotEmpty(config, relativePath);
-  validateExtends(config.extends, relativePath);
+
+  validateSchema.diagnostic(
+    ParcelConfigSchema,
+    config,
+    relativePath,
+    JSON.stringify(config, null, '\t'),
+    '@parcel/core',
+    '',
+    'Invalid Parcel Config'
+  );
+
+  /*validateExtends(config.extends, relativePath);
```

Comment:
Should probably add support for `patterns` to the validator to fully replace these

Reply:
Or maybe even create a list of all possible values to validate against? Might be a lot slower to calculate a list of all possible plugins though...

Reply:
Originally, I based the schema off of jsonschema, but as you said, regexes cannot produce helpful errors. I think a validator callback are the best solution

But we'd have to require.resolve that plugin in the validator to check if it exists so that the error is displayed at the correct position

Reply:
Would it work to just do the extra validation outside the schema validator? i.e use the schema validator for the overall structure and do additional stuff for naming and things separately?

Reply:
Than we’d have no codeframe

Reply:
Oh I mean, still throw a diagnostic just not from within the schema validator... idk if it's easier. I think TargetResolver does that in a few places.

Reply:
Experimented a bit with it and it seems like a validate callback is a pretty clean solution imo